SETUP/tests/unittests/ParamValidatorTest.php

```diff
@@ -374,9 +374,10 @@ public function testBoolFalseVariants()
 
     public function testBoolDefaultNotBool()
     {
-        // Shockingly, PHP 7.4 (at least) will not throw a TypeError if a
-        // non-boolean is passed into a function with a bool type. It instead
-        // coerces it into a bool. Maybe later versions will?
+        // PHP will not throw a TypeError if a non-boolean is passed
+        // into a function with a bool type. It instead coerces it into a bool.
+        // This is expected and there is nothing we can do about it...
+        // As it is unlikely to change, keeping this test around for documentation.
```

Comment:
As I think this is testing a fundamental property of PHP basic types, the value of the test is very much zero and I don't think we will ever enable it. I would recommend removing it but thought I would leave this as a discussion point for reviewers in case I missed something.

Reply:
I'm fine with removing the function altogether but maybe we want to keep the comment somewhere in there for the next person who goes "wait, why aren't we testing the boolean one"?

Reply:
Sounds good, added a comment in the test about not testing the case where `$default` is not a boolean.